src/Radicle/Internal/PrimFns.hs

```diff
@@ -709,8 +709,10 @@ purePrimFns = fromList $ allDocs $
           let allMod = fromEnv e
           toImport <- case v_ of
                 Just vs -> do
-                  is <- Protolude.toList <$> traverse isAtom vs ?? toLangError (OtherError "import: must be given a vector of symbols to import")
-                  pure $ Map.restrictKeys allMod (Set.fromList is)
+                  is <- Set.fromList <$> Protolude.toList <$> traverse isAtom vs ?? toLangError (OtherError "import: must be given a vector of symbols to import")
+                  if Set.null (is Set.\\ Map.keysSet allMod)
+                    then pure $ Map.restrictKeys allMod is
+                    else throwErrorHere (OtherError "import: cannot import undefined symbols")
```

Comment:
This does not tell the user which symbols were undefined - that's something that could be added.

Reply:
Maybe:

```
                  let missing = is Set.\\ Map.keysSet allMod
                  if Set.null missing
                    then pure $ Map.restrictKeys allMod is
                    else throwErrorHere
                       $ OtherError
                       $ "import: cannot import undefined symbols: "
                         <> T.intercalate ", " (fromIdent <$> Set.toList missing)
```

Reply:
Somehow I missed this comment:

The corresponding test (in `test/spec/Radicle/Tests.hs`, around ln 560), would be:

```
    , testCase "Importing complains if imports are undefined" $ do
        let prog = [s|(module
                        {:module 'foo :doc "" :exports '[x]}
                        (def x 1))
                      (import foo '[b a])
                      |]
        prog `failsWith` OtherError "import: cannot import undefined symbols: a, b"
```

Reply:
Wow, thanks!

I've adjusted your suggestions slightly:
- The list of problematic imports now appears sorted in the error message to make it "more deterministic" (I think alphabetic sorts depend on environment variables related to language settings like `LC_ALL`, but I don't think that's a big issue).
- I've amended the test case to also include the case where a symbol is neither defined nor exported.